src/router-view.js

```diff
@@ -1,13 +1,34 @@
 import {Container, inject} from 'aurelia-dependency-injection';
 import {ViewSlot, ViewLocator, customElement, noView, BehaviorInstruction} from 'aurelia-templating';
+import {bindable} from 'aurelia-framework';
 import {Router} from 'aurelia-router';
 import {Origin} from 'aurelia-metadata';
 import {DOM} from 'aurelia-pal';
 
+var timingFunctions = {
+  // animate the next view in before removing the current view;
+  before(viewSlot, view, callback) {
+    let promised = callback();
+    if (view) promised ? promised.then(()=> viewSlot.remove(view)) : viewSlot.remove(view);
```

Comment:
I think we could avoid the branch with `Promise.resolve(promised).then(() => viewSpot.remove(view))` (same for `after`)